src/client/java/io/wispforest/limelight/impl/ui/ResultEntryComponent.java

```diff
@@ -126,24 +138,32 @@ public boolean onKeyPress(int keyCode, int scanCode, int modifiers) {
 
     @Override
     public boolean onCharTyped(char chr, int modifiers) {
-        root().focusHandler().focus(screen.searchBox, FocusSource.MOUSE_CLICK);
-        screen.searchBox.onCharTyped(chr, modifiers);
-
-        return true;
+        if (root() != null) {
+            root().focusHandler().focus(screen.searchBox, FocusSource.MOUSE_CLICK);
+            screen.searchBox.onCharTyped(chr, modifiers);
+            return true;
+        }
+        return false;
     }
 
     @Override
     public void onFocusGained(FocusSource source) {
         super.onFocusGained(source);
-        surface(Surface.outline(LimelightTheme.current().focusOutlineColor()));
+        Surface s = Surface.outline(LimelightTheme.current().focusOutlineColor());
+        if (expandedChild)
+            s = s.and(Surface.flat(LimelightTheme.current().expandedResultChildEntryBackgroundColor()));
+        surface(s);
 
         applySuggestion();
     }
 
     @Override
     public void onFocusLost() {
         super.onFocusLost();
-        surface(Surface.BLANK);
+        if (expandedChild)
+            surface(Surface.flat(LimelightTheme.current().expandedResultChildEntryBackgroundColor()));
+        else
+            surface(Surface.BLANK);
         screen.searchBox.setSuggestion(null);
     }
```

Comment:
this code is a bit weird, and should probably be factored to have a `baseSurface` field that the outline is then applied to

Reply:
Sounds good

Reply:
Done